Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,87 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
-            var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+        private static void Test(string args, int expected)
+        { 
+            var ans= FindDigit(args);
+            var result = ans.Equals(expected) ? "PASS" : "FAIL";
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            string[] hlfs= new string[2];
+            string[] nums = new string[3];
+            string q="";
+            int ans =0, flag=0;
+            string sans ="";
+
+            hlfs = equation.Split("=");         //breaking the equation into lhs and rhs
+            nums[0] = hlfs[0].Split('*')[0];    // breaking the LHS into 2 nums, and Rhs into third number
+            nums[1] = hlfs[0].Split('*')[1];
+            nums[2] = hlfs[1];
+            
+            // Checking which part has the "?" mark
+
+            if(nums[0].IndexOf('?')!=-1)        
+            {
+                if(Convert.ToInt32(nums[2])%Convert.ToInt32(nums[1])==0)
+                {
+                    ans = Convert.ToInt32(nums[2])/Convert.ToInt32(nums[1]);
+                    q=nums[0];
+                }
+                else
+                    flag=1;
+            }
+            
+            else if(nums[1].IndexOf('?')!=-1)
+            {
+
+                if(Convert.ToInt32(nums[2])%Convert.ToInt32(nums[0])==0)
+                {
+                    ans = Convert.ToInt32(nums[2])/Convert.ToInt32(nums[0]);
+                    q=nums[1];
+                }
+                else
+                    flag=1;
+            }
+            
+            else if(nums[2].IndexOf('?')!=-1)
+            {
+                 ans = Convert.ToInt32(nums[0]) * Convert.ToInt32(nums[1]);
+                 q= nums[2];
+            }
+
+
+// The comparision part to find if the ? can be replaced
+            if(flag==0)
+            {
+                sans = ans.ToString();

```

Comment:
Avoid this variable, you can use ToString() at the time of calculation.

Reply:
The variable "sans" is needed multiple times in the following lines, calling .ToString() every time would cause an overhead.